### PR TITLE
Lolli tooltip

### PR DESCRIPTION
**Pull request by ede0m**

### CODE REVIEW

File: floodviz/static/js/peakflow.js [hunk 1]
```diff
@@ -53,17 +57,29 @@ document.addEventListener("DOMContentLoaded", function(event) {
...
 			.attr('class', 'bar')
 			.attr("x", function(d) {return x(d.label); })
 			.attr("y", function(d) {return y(d.value); })
 			.attr("width", x.bandwidth())
-			.attr("height", function(d) {return height - y(d.value); });
+			.attr("height", function(d) {return height - y(d.value); })
+			.on("mousemove", function(d) {
+				tooltip.transition().duration(500).style("opacity", .9);
+				tooltip.style("display", "inline-block")
+                .style("left", (d3.event.pageX) + 10 + "px")
```
Comment (ede0m) [inline floodviz/static/js/peakflow.js]: Not sure why the spacing is like this on github. But the actual file spacing is correct.

File: floodviz/static/js/peakflow.js [hunk 2]
```diff
@@ -17,13 +17,17 @@ document.addEventListener("DOMContentLoaded", function(event) {
...
-		.attr("transform", "translate(" + margin.left + "," + margin.top + ")")
-		.attr('class', 'group');
+	var svg = d3.select('#peakflow_bar').append('svg')
+				.attr("width", width + margin.left + margin.right)
+				.attr("height", height + margin.top + margin.bottom)
+				.append("g")
+				.attr("transform", "translate(" + margin.left + "," + margin.top + ")")
+				.attr('class', 'group');
+
+	var tooltip = d3.select("body")
```
Comment (skaymen) [inline floodviz/static/js/peakflow.js]: Should declarations for svg and tooltip have more consistent indentation?

File: floodviz/static/js/peakflow.js [hunk 3]
```diff
@@ -84,9 +100,19 @@ document.addEventListener("DOMContentLoaded", function(event) {
...
+				tooltip.style("display", "inline-block")
+                .style("left", (d3.event.pageX) + 10 + "px")		
+                .style("top", (d3.event.pageY) - 70 + "px")
+				.html((lolli_data.label) + "<br>" + (lolli_data.value) + " cfps");
+        	})
+    		.on("mouseout", function(d){ tooltip.style("display", "none");});
+		
 		var group = d3.select('#peakflow_bar svg .group');
-		group.append("circle").attr('class', 'cir')
+		group.append("circle")
```
Comment (skaymen) [inline floodviz/static/js/peakflow.js]: Could a more specific variable name than group be used?

File: floodviz/static/js/peakflow.js [hunk 4]
```diff
@@ -53,17 +57,29 @@ document.addEventListener("DOMContentLoaded", function(event) {
...
+			.attr("height", function(d) {return height - y(d.value); })
+			.on("mousemove", function(d) {
+				tooltip.transition().duration(500).style("opacity", .9);
+				tooltip.style("display", "inline-block")
+                .style("left", (d3.event.pageX) + 10 + "px")		
+                .style("top", (d3.event.pageY - 70) + "px")
+				.html((d.label) + "<br>" + (d.value) + " cfps");
+        	})
+    		.on("mouseout", function(d){ tooltip.style("display", "none");});
+
```
Comment (skaymen) [inline floodviz/static/js/peakflow.js]: Should the rest of code have comments like the code on lines 60-66?

File: floodviz/static/js/peakflow.js [hunk 5]
```diff
@@ -17,13 +17,17 @@ document.addEventListener("DOMContentLoaded", function(event) {
...
 	var yAxis = d3.axisLeft().scale(y).ticks(8);
 
-	var svg = d3.select('#peakflow_bar')
-		.append('svg')
-		.attr("width", width + margin.left + margin.right)
-		.attr("height", height + margin.top + margin.bottom)
-		.append("g")
-		.attr("transform", "translate(" + margin.left + "," + margin.top + ")")
-		.attr('class', 'group');
+	var svg = d3.select('#peakflow_bar').append('svg')
```
Comment (mbucknell) [inline floodviz/static/js/peakflow.js]: Consider putting .append on the next line for more consistent formatting.

File: floodviz/static/js/peakflow.js [hunk 6]
```diff
@@ -37,7 +41,7 @@ document.addEventListener("DOMContentLoaded", function(event) {
 		})
 		xAxis.tickValues(ticks);
 
-		x.domain(data.map(function(d) {return d.label; }));
+		x.domain(data.map( function(d) {return d.label; }));
```
Comment (mbucknell) [inline floodviz/static/js/peakflow.js]: Shouldn't put a space after paranthesis.

File: floodviz/static/js/peakflow.js [hunk 7]
```diff
@@ -53,17 +57,30 @@ document.addEventListener("DOMContentLoaded", function(event) {
...
 			.attr("x", function(d) {return x(d.label); })
 			.attr("y", function(d) {return y(d.value); })
 			.attr("width", x.bandwidth())
-			.attr("height", function(d) {return height - y(d.value); });
+			.attr("height", function(d) {return height - y(d.value); })
+			// tooltip event
+			.on("mousemove", function(d) {
+				tooltip.transition().duration(500).style("opacity", .9);
+				tooltip.style("display", "inline-block")
+                .style("left", (d3.event.pageX) + 10 + "px")
```
Comment (mbucknell) [inline floodviz/static/js/peakflow.js]: Indentation is wonky. These should be two lines should be lined up with .html

File: floodviz/templates/index.html [hunk 8]
```diff
@@ -4,7 +4,7 @@
 <head>
 	<meta charset="UTF-8">
 	<link rel="stylesheet" type="text/css" href="{{ url_for('static', filename='bower_components/bootstrap/dist/css/bootstrap.css') }}" />
-	<link rel="stylesheet" type="text/css" href="{{ url_for('static', filename='bower_components/bootstrap/dist/css/bootstrap.css') }}"/>>
+	<link rel="stylesheet" type="text/css" href="{{ url_for('static', filename='bower_components/bootstrap/dist/css/bootstrap.css') }}"/>
```
Comment (mbucknell) [inline floodviz/templates/index.html]: This is a duplicate of line 7 and should be removed.

**Pull request merged**